tests/contracts/ValidationRulesStorage.sol

```diff
@@ -0,0 +1,52 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity ^0.8.25;
+
+import "@account-abstraction/contracts/interfaces/IEntryPoint.sol";
+import "./ITestAccount.sol";
+
+contract ValidationRulesStorage is IState {
+    IEntryPoint public entryPoint;
+    uint256 public state;
+
+    function funTSTORE() external override returns(uint256) {
+        assembly {
+            tstore(0, 1)
+        }
+        return 0;
+    }
+
+    function funSSTORE() external returns(uint256) {
+        assembly {
+            sstore(0, 1)
+        }
+        return 0;
+    }
+
+
+    function funTLOAD() external returns(uint256) {
+        uint256 tval;
+        assembly {
+            tval := tload(0)
+        }
+        emit State(tval, tval);
```

Comment:
why emit in test TLOAD?

Reply:
I think the point of emitting anything in tests is to prevent the optimizer from removing the entire chunk of code when it's results don't have any observable side effects. And for us, the side effects lie in the tracing results alone, which the optimizer doesn't know.

Reply:
i don't think the optimizer removes assembly code.

Reply:
I believe I saw this happen and it is very hard to figure out what's wrong when it happens. Emitting an event helps both prevent the optimisations, and helps debugging the code if something does go wrong.